src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -1817,107 +1817,123 @@ Node TheoryStringsRewriter::rewriteContains( Node node ) {
 
 Node TheoryStringsRewriter::rewriteIndexof( Node node ) {
   Assert(node.getKind() == kind::STRING_STRIDOF);
-  std::vector< Node > children;
-  getConcat( node[0], children );
-  //std::vector< Node > children1;
-  //getConcat( node[1], children1 );  TODO
-  std::size_t start = 0;
-  std::size_t val2 = 0;
-  if( node[2].isConst() ){
+  NodeManager* nm = NodeManager::currentNM();
+
+  std::vector<Node> children0;
+  getConcat(node[0], children0);
+  if (children0[0].isConst() && node[1].isConst() && node[2].isConst())
+  {
+    CVC4::String t = node[1].getConst<String>();
+    if (t.size() == 0)
+    {
+      Node negone = nm->mkConst(Rational(-1));
+      return returnRewrite(node, negone, "idof-empty");
+    }
     CVC4::Rational RMAXINT(LONG_MAX);
-    if( node[2].getConst<Rational>()>RMAXINT ){
+    if (node[2].getConst<Rational>() > RMAXINT)
```

Comment:
This is a really tricky point.

Our string constants cannot store more than LONG_MAX characters, hence we know that, due to this limitation, that the index is out of bounds.

However, it would be unsound to apply this rewrite in general.  Actually concretely you could imagine:

str.index( str.++( x, "A" ), "A", 99999999999999999999999999999999999999999999999999999999999999999999) = 0 ^
str.len( x ) = 99999999999999999999999999999999999999999999999999999999999999999999